### ISSUE TITLE
3P Media Categories Accessibility

### ISSUE THREAD

**Issue by swissspidy** [GoogleForCreators/web-stories-wp#3398]
The Media3p categories list as it exists today is not fully accessible by keyboard.

Thankfully @jhtin added some `aria-*` attributes, which is great. However, there are still issues.

* The pills themselves are just `<span>` elemens, and need to be proper `button` elements so they can receive focus.
* Since the categories list is a **tablist**, navigation within the list should happen via arrow keys, and pressing <kbd>tab</kbd> would navigate out of the list to the next focusable element (the arrow button in this case)  
  See the tabs at the top of the library panel for inspiration.
* The wrapping element (`CategorySection`) has an `aria-expanded` attribute, but is not a `section`. However, since it's actually controlled by the arrow button, the `aria-expanded` attribute should be on the button
* The arrow button needs an `aria-controls` attribute that contains the ID of `CategorySection`

**How can this be tested?**

We have `jest-axe` in the repo, so for Jest tests we could do something like this:

```js
import { axe } from 'jest-axe';

it('should render categories without accessibility violations', async () => {
  const { container } = renderWithTheme(
    <Media3pCategories
      categories={categories}
      selectedCategoryName={undefined}
      selectCategory={selectCategoryMock}
      deselectCategory={deselectCategoryMock}
    />
  );

  const results = await axe(container);
  expect(results).toHaveNoViolations();
});
```

Note: we might need to [tweak the axe configuration](https://github.com/nickcolley/jest-axe#axe-configuration).

---

_Do not alter or remove anything below. The following sections will be managed by moderators only._
